client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -872,6 +897,30 @@ def get_template_preset(self):
             "create_first_version": create_first_version
         }
 
+    def trigger_event(self, topic, data=None, source=None):
+        self._event_system.emit(topic, data, source)
+
+    def register_event_callback(self, topic, callback, order=None):
+        self._event_system.add_callback(topic, callback, order=order)
+
+    def register_on_finished_callback(
+            self, callback, order=None
+    ):
+        self.register_event_callback(
+            topic="template.finished",
+            callback=callback,
+            order=order
+        )
+
+    def register_on_depth_processed_callback(
+            self, callback, order=None
+    ):
+        self.register_event_callback(
+            topic="template.depth_processed",
+            callback=callback,
+            order=order
+        )
```

Comment:
Should this be `emit_event` instead of `trigger_event` and `add_*_callback` instead of `register_*_callback` to match more with the method names on the actual event system @iLLiCiTiT ?